server/push_test.go

```diff
@@ -240,11 +246,39 @@ func TestShouldUpdateLastUsed(t *testing.T) {
 	assert.True(lu.Before(s.rtmpConnections["mani1"].lastUsed))
 }
 
+func ignoreRoutines() []goleak.Option {
+	funcs2ignore := []string{"github.com/golang/glog.(*loggingT).flushDaemon", "go.opencensus.io/stats/view.(*worker).start",
+		"github.com/rjeczalik/notify.(*recursiveTree).dispatch", "github.com/rjeczalik/notify._Cfunc_CFRunLoopRun", "github.com/ethereum/go-ethereum/metrics.(*meterArbiter).tick",
+		"github.com/ethereum/go-ethereum/consensus/ethash.(*Ethash).remote", "github.com/ethereum/go-ethereum/core.(*txSenderCacher).cache",
+		"internal/poll.runtime_pollWait", "github.com/livepeer/go-livepeer/core.(*RemoteTranscoderManager).Manage", "github.com/livepeer/lpms/core.(*LPMS).Start",
+		"github.com/livepeer/go-livepeer/server.(*LivepeerServer).StartMediaServer", "github.com/livepeer/go-livepeer/core.(*RemoteTranscoderManager).Manage.func1",
+		"github.com/livepeer/go-livepeer/server.(*LivepeerServer).HandlePush.func1", "github.com/rjeczalik/notify.(*nonrecursiveTree).dispatch",
+		"github.com/rjeczalik/notify.(*nonrecursiveTree).internal"}
+
+	res := make([]goleak.Option, 0, len(funcs2ignore))
+	for _, f := range funcs2ignore {
+		res = append(res, goleak.IgnoreTopFunction(f))
+	}
+	return res
+}
+
+func serverCleanup(s *LivepeerServer) {
```

Comment:
Can this be moved into `mediaserver_test.go` along the other LivepeerServer helpers such as `setupServerWithCancel` ?

Reply:
While this type of cleanup is likely the correct thing to do, curious what the effects are of not cleaning up anything? Our tests seem to have been running OK without this.

Reply:
Yeah, I've started to writing this clean function trying to shutdown everything correctly, but then realised that there is tons of goroutines started over which I don't have control. I'll just remove this function, anyway I was forced to make long list (`funcs2ignore`) of goroutines to ignore.

Reply:
Actually I was to quick to write previous comment - this function is needed because other tests too starts this goroutine for which we're checking. Added comment.